static/app/components/profiling/profilesTable.tsx

```diff
@@ -0,0 +1,279 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import {SectionHeading} from 'sentry/components/charts/styles';
+import DateTime from 'sentry/components/dateTime';
+import GridEditable, {
+  COL_WIDTH_UNDEFINED,
+  GridColumnOrder,
+} from 'sentry/components/gridEditable';
+import ProjectBadge from 'sentry/components/idBadge/projectBadge';
+import Link from 'sentry/components/links/link';
+import Pagination from 'sentry/components/pagination';
+import PerformanceDuration from 'sentry/components/performanceDuration';
+import {IconCheckmark, IconClose} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Trace} from 'sentry/types/profiling/core';
+import {defined} from 'sentry/utils';
+import {Container, NumberContainer} from 'sentry/utils/discover/styles';
+import {getShortEventId} from 'sentry/utils/events';
+import {
+  generateFlamegraphSummaryRoute,
+  generateFunctionsRouteWithQuery,
+} from 'sentry/utils/profiling/routes';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+
+const REQUIRE_PROJECT_COLUMNS = new Set(['id', 'project_id', 'transaction_name']);
+
+interface ProfilesTableProps {
+  error: string | null;
+  isLoading: boolean;
+  traces: Trace[];
+  columnOrder?: TableColumnKey[];
+  hideHeader?: boolean;
+  pageLinks?: string | null;
+}
+
+function ProfilesTable(props: ProfilesTableProps) {
+  const location = useLocation();
+
+  return (
+    <Fragment>
+      {!props.hideHeader && (
```

Comment:
Can we make header a component prop here, then instead of passing hideHeader we just dont pass a header component? I usually avoid hideX/showX props and opt for more composable patterns, that way we can also render whatever header we'd want in the future (if we need to) and have more flexibility going forward